<launch>
  <!-- 参数设置 -->
  <arg name="shared_images_dir" default="/shared/images"/>
  <arg name="shared_results_dir" default="/shared/results"/>
  <arg name="image_topic" default="/usb_cam/image_raw"/>
  <arg name="save_frequency" default="1.0"/>
  <arg name="check_frequency" default="2.0"/>
  <arg name="max_images" default="100"/>
  
  <!-- 启动USB摄像头 -->
  <include file="$(find usb_cam)/launch/usb_cam-test.launch"/>
  
  <!-- 启动图像保存节点 -->
  <node name="image_saver" pkg="view_detect" type="image_saver.py" output="screen">
    <param name="shared_dir" value="$(arg shared_images_dir)"/>
    <param name="image_topic" value="$(arg image_topic)"/>
    <param name="save_frequency" value="$(arg save_frequency)"/>
    <param name="max_images" value="$(arg max_images)"/>
  </node>
  
  <!-- 启动结果发布节点 -->
  <node name="result_publisher" pkg="view_detect" type="result_publisher.py" output="screen">
    <param name="results_dir" value="$(arg shared_results_dir)"/>
    <param name="check_frequency" value="$(arg check_frequency)"/>
  </node>
</launch> 